x/metrix/module.go

```diff
@@ -153,16 +166,20 @@ func (am AppModule) ExportGenesis(ctx sdk.Context, cdc codec.JSONCodec) json.Raw
 func (AppModule) ConsensusVersion() uint64 { return 1 }
 
 // BeginBlock executes all ABCI BeginBlock logic respective to the capability module.
-func (am AppModule) BeginBlock(ctx sdk.Context, _ abci.RequestBeginBlock) {}
+func (am AppModule) BeginBlock(ctx context.Context) error { return nil }
 
 // EndBlock executes all ABCI EndBlock logic respective to the capability module. It
 // returns no validator updates.
-func (am AppModule) EndBlock(ctx sdk.Context, _ abci.RequestEndBlock) []abci.ValidatorUpdate {
-	if ctx.BlockHeight()%cUpdateUptimeBlockInterval == 0 {
+func (am AppModule) EndBlock(ctx context.Context) error {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	if sdkCtx.BlockHeight()%cUpdateUptimeBlockInterval == 0 {
 		am.keeper.PurgeRelayMetrics(ctx)
 		am.keeper.UpdateRelayMetrics(ctx)
-		am.keeper.UpdateUptime(ctx)
+		err := am.keeper.UpdateUptime(ctx)
```

Comment:
I don't think this needs to return an error. The current definition doesn't, and this is really just a fire & forget mechanism to track what's been going on. Logging the errors in `UpdateUpdate` is enough, we dont' need to bubble them up here.

Reply:
Okay